### PR TITLE
Remove Unused Materials button

### PR DESCRIPTION
**Pull request by zrksyd**
I'm not sure if I made a complete list of blacklisted materials. I also wasn't sure whether the button should default to model.numdlb or have the user choose which NUMDLB but either way.

### CODE REVIEW

File: src/editors/matl.rs [hunk 1]
```diff
@@ -194,6 +194,31 @@ fn remove_duplicates(entries: &mut Vec<MatlEntryData>) {
...
+        String::from("EyeLD"), String::from("EyeLD1"), String::from("EyeLD2"),
+        String::from("EyeRD"), String::from("EyeRD1"), String::from("EyeRD2"),
+        String::from("EyeLG"), String::from("EyeLG1"), String::from("EyeLG2"),
+        String::from("EyeRG"), String::from("EyeRG1"), String::from("EyeRG2"),
+        String::from("EyeLL"), String::from("EyeLL1"), String::from("EyeLL2"),
+        String::from("EyeRL"), String::from("EyeRL1"), String::from("EyeRL2"),
+    ];
+
+    let mut visited = Vec::with_capacity(modl_entries.len() + blacklist.len());
+    visited.append(&mut blacklist);
```
Comment (ScanMountGoat) [inline src/editors/matl.rs:210]: You can just initialize the visited vector with these entries with `let mut visited = vec![...];` The list is small, so I would just focus on readability and not worry about reallocations when inserting.

File: src/editors/matl.rs [hunk 2]
```diff
@@ -194,6 +194,31 @@ fn remove_duplicates(entries: &mut Vec<MatlEntryData>) {
...
+
+    let modl_iter = modl_entries.iter();
+    for modl_entry in modl_iter {
```
Comment (ScanMountGoat) [inline src/editors/matl.rs:213]: This can just be `for model_entry in &model_entries {`.

**Comment by ScanMountGoat** [comment]
This sounds like a good idea. We'll need to think about more about some of the edge cases. Is there a reason you wanted users to select a different model.numdlb instead of just using the model.numdlb from the current folder?

File: src/editors/matl.rs (hunk 1 above)
Comment (ScanMountGoat) [reply src/editors/matl.rs:210]: I would also add a comment explaining why these material names are important.

**Comment by zrksyd** [comment]
I didn't default to model.numdlb because there were no other menu buttons I referenced that defaulted to model.XXX so I was referencing at the "match to order" buttons to be consistent.

**Comment by ScanMountGoat** [comment]
I would just add a `modl: Option<&ModlData>` parameter to the `menu_bar` function. Most model folders should already have a numdlb loaded. There are three approaches to handling the case where `modl` is `None`.

1. Just do nothing, which is going to be confusing for the user.
2. Disable the menu option if the modl is missing. The dots would be removed from the menu option since it doesn't open an additional dialog.
3. Prompt the user to select a model.numdlb.

The first option is not a great idea. I prefer the second option, but you can implement the third option if you prefer. If you choose the third option, I recommend trying to set the title on the file dialog to something like "Select a model.numdlb".

**Comment by zrksyd** [comment]
The second option is a better user experience in my opinion, but I had trouble implementing it so I left it as a to-do comment for now.

**Comment by ScanMountGoat** [comment]
You should be able to pass in the modl and use `ui.add_enabled`. 
```rust
if ui.add_enabled(modl.is_some(), egui::Button::new("Remove Unused Materials")).clicked() {
    ui.close_menu();

    if let Some(modl) = modl {
        ...
    }
}
```

**Comment by zrksyd** [comment]
Only app.rs uses find_file and find_file_mut and I don't think it would be such a good idea to copy that over for something relatively simple.

**Comment by ScanMountGoat** [comment]
The modl is already passed into the matl editor. You just need to add a parameter to the menu_bar function.

**Comment by zrksyd** [comment]
```
error[E0382]: use of moved value: `modl`
  --> src\editors\matl.rs:73:27
   |
58 |               saved |= menu_bar(ui, matl, modl, ui_state, material_presets, folder_name, file_name);
   |                                           ---- value moved here
...
73 |                   .show(ui, |ui| {
   |  ___________________________^
74 | |                     changed |= edit_matl_entries(
75 | |                         ctx,
76 | |                         ui,
...  |
86 | |                     );
87 | |                 });
   | |_________________^ value used here after move
   |
   = note: move occurs because `modl` has type `std::option::Option<&mut ssbh_data::modl_data::ModlData>`, which does not implement the `Copy` trait
   ```

**Comment by ScanMountGoat** [comment]
Oh the code I wrote wants to take ownership of `modl`. If you want to borrow `modl` to allow it to be used again, you need a reference like `modl: &Option<&mut ModlData>`. This is one of the harder Rust concepts to learn. There's a reference on the rust docs if you're curious. https://doc.rust-lang.org/book/ch04-01-what-is-ownership.html#ownership-and-functions

**Pull request merged**